src/plugins/nginx.go

```diff
@@ -569,40 +569,40 @@ func (n *Nginx) monitor(processInfo []core.Process) error {
 
 	logErrorChannel := make(chan string, len(errorLogs))
 	pidsChannel := make(chan string)
+	defer close(logErrorChannel)
+	defer close(pidsChannel)
 
 	go n.monitorLogs(errorLogs, logErrorChannel)
 	go n.monitorPids(processInfo, pidsChannel)
 
-	for i := 0; i < 2; i++ {
+	for i := 0; i < (1 + len(errorLogs)); i++ {
```

Comment:
If I understand correctly `1 + len(errorLogs)` is because we always expect exactly one msg from the `pidsChannel` and a message from every entry in `errorLogs`/`errorLogsChannel`.

Might be worth adding a comment to make that clear?